sdk/schemaregistry/azure-schemaregistry-avroserializer/azure/schemaregistry/serializer/avroserializer/_schema_registry_avro_serializer.py

```diff
@@ -141,7 +131,7 @@ def serialize(self, value, schema, **kwargs):
             cached_schema = parsed_schema
 
         record_format_identifier = b"\0\0\0\0"
-        schema_id = self._get_schema_id(cached_schema.fullname, cached_schema, **kwargs)
+        schema_id = self._get_schema_id(cached_schema.fullname, str(cached_schema), **kwargs)
```

Comment:
why cast `cached_schema` to be type of string here?

----

I'm thinking of the accepted type of input `schema` here based on the discussion we just had today -- hide apache avro from the surfaces areas.

maybe we could start by just supporting "str" first, because accepting bytes brings the problem of decoding bytes into string.

do we know what type fastavro is expecting for the schema, string, bytes or dict?

Reply:
`lru_cache` [does not work when the parameters are mutable](https://stackoverflow.com/questions/31235731/how-to-apply-functools-lru-cache-to-function-with-mutable-parameters) (i.e., `avro.schema.Schema`), and we cast `schema` into a string in `_get_schema-id` anyway and don't every use the actual `Schema` object for anything.
___
I think removing `Schema` as an input type is a good idea, but just wondering if this is part of the surface? since this `serialize` method only accepts `str/bytes` and it's only internally that we convert/pass around `avro.schema.Schema`.

- `fastavro` expects dicts for schema. (sample code here: https://github.com/Azure/azure-sdk-for-python/issues/20708#issuecomment-924460514)
- should we check with the rest of the feature crew about only supporting `str` for now instead of `bytes` + `str`?

Reply:
only support `str` as this aligns with other languages. This will already be addressed in this PR: https://github.com/Azure/azure-sdk-for-python/pull/20763/files.
Merge this PR after.

Reply:
keeping everything, except for `bytes` as a type for `schema`. that will be addressed in another PR which will be merged first. https://github.com/Azure/azure-sdk-for-python/pull/20763